common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -136,11 +142,12 @@ private FetchShuffleBlocks createFetchShuffleBlocksMsg(
         mapIdToReduceIds.get(mapId).add(Integer.parseInt(blockIdParts[4]));
       }
     }
-    long[] mapIds = Longs.toArray(mapIdToReduceIds.keySet());
+    long[] mapIds = Longs.toArray(orderedMapId);
```

Comment:
If we use a LinkedHashMap for `mapIdToReduceIds`, then `mapIdToReduceIds.keySet` will return the mapIds in the same order as they were inserted. So the order will be same as blockIds.

Reply:
This assumes the block ids are ordered by map id and reduce id. If the block ids are `shuffle_1_1_1, shuffle_1_2_1, shuffle_1_1_2`, then I don't think `FetchShuffleBlocks` can retain the order anymore.

Reply:
That's right so we can not just use LinkedHashMap and maintain a separate arrayList of mapIds.

Reply:
ok. In that case, we could recompute the member `blockIds`  from the `FetchShuffleBlocks`. I think that was the initial fix.